### ISSUE TITLE
XGBoost appears to be incompatible with libomp 12.0

### ISSUE THREAD

**Issue by hcho3** [dmlc/xgboost#7106]
I tried running Python tests on my Macbook Pro (16-inch, 2019) and got an segfault:
```
$ PYTHONPATH=./python-package/ python -m pytest -v -x tests/python/test_basic.py::TestBasic::test_basic
====================================================================== test session starts =======================================================================
platform darwin -- Python 3.7.10, pytest-6.2.4, py-1.10.0, pluggy-0.13.1 -- /usr/local/Caskroom/miniconda/base/envs/cpu_test/bin/python
cachedir: .pytest_cache
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/phcho/Desktop/xgboost/.hypothesis/examples')
rootdir: /Users/phcho/Desktop/xgboost/tests, configfile: pytest.ini
plugins: cov-2.12.1, hypothesis-6.14.2
collected 1 item                                                                                                                                                 

tests/python/test_basic.py::TestBasic::test_basic Fatal Python error: Segmentation fault

Thread 0xSegmentation fault: 11
```

How to reproduce:
* Upgrade `libomp` by running `brew install libomp`. My machine has `libomp` 12.0.1.
* Compile the latest source of XGBoost: 
```
mkdir build
cd build
cmake .. -GNinja -DCMAKE_BUILD_TYPE=Debug 
ninja
```
(The same error happens when using the Release build.)
* Run pytest with `PYTHONPATH=./python-package/ python -m pytest -v -x tests/python/test_basic.py::TestBasic::test_basic`

The segfault error went away when I downgraded libomp to 11.1.0, using tricks documented in https://remarkablemark.org/blog/2017/02/03/install-brew-package-version/

**Comment by hcho3** [comment]
To unblock our CI, let's use libomp 11.1.0 for all MacOS tests.

**Comment by hcho3** [comment]
Closing as duplicate of #7039
